packages/ckeditor5-html-support/src/integrations/image.js

```diff
@@ -0,0 +1,222 @@
+			const viewContainerElement = viewImageElement.parent;
+
+			preserveElementAttributes( viewImageElement, 'htmlAttributes' );
+
+			if ( viewContainerElement.is( 'element', 'figure' ) ) {
+				preserveElementAttributes( viewContainerElement, 'htmlFigureAttributes' );
+			} else if ( viewContainerElement.is( 'element', 'a' ) ) {
+				preserveLinkAttributes( data, preserveElementAttributes, viewContainerElement );
+			}
+
+			function preserveElementAttributes( viewElement, attributeName ) {
+				const viewAttributes = dataFilter._consumeAllowedAttributes( viewElement, conversionApi );
+
+				if ( viewAttributes ) {
+					conversionApi.writer.setAttribute( attributeName, viewAttributes, data.modelRange );
+				}
+			}
+		}, { priority: 'low' } );
+	};
+
+	// For a block image, we want to preserve the attributes on our own.
+	// The inline image attributes will be handled by the GHS automatically.
+	function preserveLinkAttributes( data, preserveElementAttributes, viewContainerElement ) {
+		if ( data.modelRange && data.modelRange.getContainedElement().is( 'element', 'imageBlock' ) ) {
+			preserveElementAttributes( viewContainerElement, 'htmlLinkAttributes' );
+		}
+
+		// If we're in a link, then the `<figure>` element should be one level higher.
+		if ( viewContainerElement.parent.is( 'element', 'figure' ) ) {
+			preserveElementAttributes( viewContainerElement.parent, 'htmlFigureAttributes' );
+		}
+	}
+}
+
+// A model-to-view conversion helper applying attributes from the {@link module:image/image~Image Image}
+// feature.
+//
+// @private
+// @returns {Function} Returns a conversion callback.
+function modelToViewImageAttributeConverter() {
+	return dispatcher => {
+		addInlineAttributeConversion( 'htmlAttributes' );
+
+		addBlockAttributeConversion( 'img', 'htmlAttributes' );
+		addBlockAttributeConversion( 'figure', 'htmlFigureAttributes' );
+		addBlockImageLinkAttributeConversion();
+
+		function addInlineAttributeConversion( attributeName ) {
+			dispatcher.on( `attribute:${ attributeName }:imageInline`, ( evt, data, conversionApi ) => {
+				if ( !conversionApi.consumable.consume( data.item, evt.name ) ) {
+					return;
+				}
+
+				const viewElement = conversionApi.mapper.toViewElement( data.item );
+
+				setViewAttributes( conversionApi.writer, data.attributeNewValue, viewElement );
+			}, { priority: 'low' } );
+		}
+
+		function addBlockAttributeConversion( elementName, attributeName ) {
+			dispatcher.on( `attribute:${ attributeName }:imageBlock`, ( evt, data, conversionApi ) => {
+				if ( !conversionApi.consumable.consume( data.item, `attribute:${ attributeName }:imageBlock` ) ) {
+					return;
+				}
+
+				const containerElement = conversionApi.mapper.toViewElement( data.item );
+				const viewElement = getDescendantElement( conversionApi, containerElement, elementName );
+
+				// The viewElement can be empty, e.g. elementName is `a`, but on editing downcast we won't find `a`.
+				/* istanbul ignore next */
```

Comment:
As discussed. Consider adding tests instead of ignoring.

Reply:
Removed the condition entirely, as it happened to be outdated.